contracts/options.cairo

```diff
@@ -162,6 +162,9 @@ func mint_option_token{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_che
             assert contract_option_side = option_side;
         }
 
+        let (user_address) = get_caller_address();
+        let (optsize_before) = IOptionToken.balanceOf(contract_address=option_token_address, account=user_address);
```

Comment:
I think it should be checked against `option_size` and not the balance... the user can go around by having multiple addresses... so neither is fool proof, but I prefer the option_size

if you disagree, just comment that you disagree